src/logger/src/logger.rs

```diff
@@ -1,428 +1,155 @@
         }
 
-        format!("[{}]", prefix.join(":"))
-    }
-
-    /// if the max level hasn't been configured yet, set it to default
-    fn try_init_max_level(&self) {
-        // if the max level hasn't been configured yet, set it to default
-        if max_level() == LevelFilter::Off {
-            self.set_max_level(DEFAULT_MAX_LEVEL);
+        if let Some(show_log_origin) = config.show_log_origin {
+            guard.format.show_log_origin = show_log_origin;
         }
-    }
-
-    /// Preconfigure the logger prior to initialization.
-    /// Performs the most basic steps in order to enable the logger to write to stdout or stderr
-    /// even before calling LOGGER.init(). Calling this method is optional.
-    /// This function can be called any number of times before the initialization.
-    /// Any calls made after the initialization will result in `Err()`.
-    ///
-    /// # Arguments
-    ///
-    /// * `instance_id` - Unique string identifying this logger session. This id is temporary and
-    ///   will be overwritten upon initialization.
-    ///
-    /// # Example
-    ///
-    /// ```
-    /// use std::ops::Deref;
-    ///
-    /// use logger::LOGGER;
-    ///
-    /// LOGGER
-    ///     .deref()
-    ///     .configure(Some("MY-INSTANCE".to_string()))
-    ///     .unwrap();
-    /// ```
-    pub fn configure(&self, instance_id: Option<String>) -> Result<()> {
-        self.init
-            .call_init(|| {
-                if let Some(some_instance_id) = instance_id {
-                    self.set_instance_id(some_instance_id);
-                }
-
-                self.try_init_max_level();
 
-                // don't finish the initialization
-                false
-            })
-            .map_err(LoggerError::Init)
-    }
-
-    /// Initialize log system (once and only once).
-    /// Every call made after the first will have no effect besides returning `Ok` or `Err`.
-    ///
-    /// # Arguments
-    ///
-    /// * `header` - Info about the app that uses the logger.
-    /// * `log_dest` - Buffer for plain text logs. Needs to implements `Write` and `Send`.
-    ///
-    /// # Example
-    ///
-    /// ```
-    /// use std::io::Cursor;
-    ///
-    /// use logger::LOGGER;
-    ///
-    /// let mut logs = Cursor::new(vec![0; 15]);
-    ///
-    /// LOGGER.init("Running Firecracker v.x".to_string(), Box::new(logs));
-    /// ```
-    pub fn init(&self, header: String, log_dest: Box<dyn Write + Send>) -> Result<()> {
-        self.init
-            .call_init(|| {
-                let mut g = extract_guard(self.log_buf.lock());
-                *g = log_dest;
-
-                self.try_init_max_level();
-
-                // finish init
-                true
-            })
-            .map_err(LoggerError::Init)?;
-
-        self.write_log(&header);
+        // Ensure we drop the guard before attempting to log, otherwise this
+        // would deadlock.
+        drop(guard);
+        if target_changed {
```

Comment:
Maybe add a comment here with reasons we print `"Running Firecracker v{FIRECRACKER_VERSION}"` every time we change target.